src/features/AnnouncementModal/AnnouncementModal.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Flex,
+  Image,
+  Modal,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  ModalBody,
+  Stack,
+  Link,
+} from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+import { ProgressIndicator } from "components/ProgressIndicator"
+import { useCallback, useMemo, useState } from "react"
+import { BiRightArrowAlt } from "react-icons/bi"
+
+import { useAnnouncements } from "hooks/useAnnouncement"
+
+import { Announcement } from "types/announcements"
+
+import { NewFeatureTag } from "./components/NewFeatureTag"
+
+interface AnnouncementModalProps {
+  isOpen: boolean
+  onClose: () => void
+  announcements: Announcement[]
+  link: string
+}
+
+export const AnnouncementModal = ({
+  isOpen,
+  announcements,
+  onClose,
+  link,
+}: AnnouncementModalProps): JSX.Element => {
+  const [currActiveIdx, setCurrActiveIdx] = useState<number>(0)
+  const { setLastSeenAnnouncement } = useAnnouncements()
+  const isLastAnnouncement = useMemo(
+    () => currActiveIdx === announcements.length - 1,
+    [announcements.length, currActiveIdx]
+  )
+
+  const handleNextClick = useCallback(() => {
+    if (isLastAnnouncement) {
+      onClose()
+      setLastSeenAnnouncement()
+    }
+
+    setCurrActiveIdx(Math.min(currActiveIdx + 1, announcements.length - 1))
+  }, [
+    announcements.length,
+    currActiveIdx,
+    isLastAnnouncement,
+    onClose,
+    setLastSeenAnnouncement,
+  ])
+
+  const { title, description, image, tags } = announcements[currActiveIdx]
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} size="md">
+      <ModalOverlay />
+      <ModalContent>
+        <ModalCloseButton />
+        <Image bg="base.canvas.brandLight" pt="4.5rem" as={image} />
+        <ModalHeader>
+          {tags.map((tagVariant) => {
+            switch (tagVariant) {
+              case "New Feature":
+                return <NewFeatureTag />
+              default: {
+                const unimplVariant: never = tagVariant
+                throw new Error(
+                  `Unimplemented tag variant found: ${unimplVariant}`
```

Comment:
will this break the page?

Reply:
yes but this should be of type `never` and the intention is to catch this at compile time (since you can never assign a concrete value to `nev`r)`

Reply:
just to confirm, we dont need to implement the case where there are no tags correct?

Reply:
wouldn't a console.error suffice?

Reply:
@kishore03109 nope, cos nat say cfm will have tags 

@harishv7 yes but better to break in local dev than for it to be silently ignored @_@